### DISCUSSION TITLE
トラックボールが反応しない

### DISCUSSION THREAD

**Discussion by palchan-net** [Yowkees/keyball#334] [Q&A]
度々ですみません。トラックボールが反応しない現象が出ています。
![image](https://github.com/Yowkees/keyball/assets/143974554/f0fbd92a-474d-47cf-93c3-9f0f8073bf6d)
![image](https://github.com/Yowkees/keyball/assets/143974554/5ff561cd-3501-437f-89e4-f3c8db801e2f)

まずトラックボールは、特定のドライバーが必要なものでしょうか。（Windows11、キーボードのファームウェア書き込みのみ実施の状態）
一応トラックボール周辺の半田付けし直しはしてみたのですが現象変わらずで、それ以外の箇所で半田付けし直しが必要なのか、つけ直してもなお状態がよくないのかというところをご教示いただけますと助かります。

**Comment by Yowkees** [comment]
キー入力ができるようであればドライバー等は関係ないかと思います。
トラックボールを動かしたときにOLEDに表示されている移動量は変化しますか？

少々画像が荒いようで、トラックボール基板の半田付け状況は確認できませんでしたので、再度お送りいただければ幸いです。

**Reply by palchan-net** [reply]
回答ありがとうございます。
OLEDは2行目のballの列の値でしょうか。0,0,0,0のまま変化ありません。
![image](https://github.com/Yowkees/keyball/assets/143974554/940bcc3e-87ab-493f-8ee5-3b2b934d082c)

トラックボール基板について、見やすいように角度をつけたものも含めて再掲いたします。
接写で少し潰れてしまうようですがこちらで確認できますでしょうか。
![image](https://github.com/Yowkees/keyball/assets/143974554/1d5b1af6-3a1b-47b9-9387-c002f37a4809)
![image](https://github.com/Yowkees/keyball/assets/143974554/a606ddd1-53d1-4a33-bb3e-cff0ab97c50d)
![image](https://github.com/Yowkees/keyball/assets/143974554/78dfc897-1563-4983-9e1e-e869ed5f77a7)

接続時に浮いてはないと思っておりますが、こちらも念のため送付します。
![image](https://github.com/Yowkees/keyball/assets/143974554/7b27301c-cd52-483a-ba2c-fee263f95664)
![image](https://github.com/Yowkees/keyball/assets/143974554/8e9b5fa6-acba-4029-9741-17625fb12d5b)

**Reply by palchan-net** [reply]
ProMicro裏も念のため送付します。
![image](https://github.com/Yowkees/keyball/assets/143974554/f6d3ad00-09af-4491-81fa-51d74c046dc0)

**Reply by palchan-net** [reply]
切り分けで別途購入したKeyball61を組み立てて、問題が起きているKeyball39のセンサーをはめて動作試験してみたところ動きました。
![image](https://github.com/Yowkees/keyball/assets/143974554/d5e557b4-8aed-47fd-a132-7872c056134a)

トラックボール基板は問題がなく、キーボードメイン基板側の問題ということで考えて良さそうです。
確認した方が良い場所などありましたらご教示いただけますと助かります。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

トラックボール基板には問題がないということで、ご調査ありがとうございました。

お写真を見まして、下記のProMicroの半田付けですが、ちょっと半田の量が少ないかなと思いました。
ちょうど丸をした端子に工学センサーの信号線が含まれていますので、半田付けを見直していただければ幸いです。

![KB39_trackballfix](https://github.com/Yowkees/keyball/assets/84209772/466437d8-0dd1-41bc-8c85-dc0b37763418)

**Reply by palchan-net** [reply]
回答ありがとうございます。裏側だけ一所懸命つけ直しとかしていて表側を失念しておりました。
ご指摘いただいたところ含めて怪しいところを一通りつけ直したら動くようになりました。
つけ直し等しているうちに半田付けスキルが向上したのでKeyball61は比較的エラー無く進みましたが、初期の半田付けはイマイチな箇所が多かったようです（Keyball61もLEDで悶絶している最中なのでまだまだですが）

これでKeyball39は無事完成です。ご対応いただき誠にありがとうございました。
